src/sync.d

```diff
@@ -445,9 +445,36 @@ final class SyncEngine
 			// Check if there is an interrupted upload session
 			if (session.restore()) {
 				log.log("Continuing the upload session ...");
+				string sessionFilePath = session.getSessionFilePath();
 				auto item = session.upload();
-				saveItem(item);
-			}		
+				
+				if (!uploadFailed) {
```

Comment:
Is `uploadFailed` ever going to be true here?

Reply:
If an upload fails, using the functions that session.upload() utilities , there is the potential for this to be flagged as true.

Reply:
Resolved by https://github.com/abraunegg/onedrive/pull/1399/commits/9cdf81b074715c9f07a5be92d99d7bf5c22cc324